src/fe-text/statusbar-config.c

```diff
@@ -334,98 +354,137 @@ static void cmd_statusbar_list(void)
+		"statusbar add", &optlist, &name))
+		return;
 
-/* SYNTAX: STATUSBAR <name> RESET */
-static void cmd_statusbar_reset(const char *data, void *server,
-				void *item, CONFIG_NODE *node)
-{
-	CONFIG_NODE *parent;
+	if (*name == '\0') {
+		cmd_param_error(CMDERR_NOT_ENOUGH_PARAMS);
+	}
 
-	parent = iconfig_node_traverse("statusbar", TRUE);
-	parent = iconfig_node_section(parent, active_statusbar_group->name,
-				     NODE_TYPE_BLOCK);
+	error = 0;
+
+	type = NULL;
+	data = g_hash_table_lookup(optlist, "type");
+	if (data != NULL) {
+		if (g_ascii_strcasecmp(data, "window") == 0)
+			type = "window";
+		else if (g_ascii_strcasecmp(data, "root") == 0)
+			type = "root";
+		else {
+			printformat(NULL, NULL, MSGLEVEL_CLIENTERROR,
+				    TXT_STATUSBAR_UNKNOWN_TYPE, data);
+			error++;
+		}
+	}
 
-        iconfig_node_set_str(parent, node->key, NULL);
-}
+	placement = NULL;
+	data = g_hash_table_lookup(optlist, "placement");
+	if (data != NULL) {
+		if (g_ascii_strcasecmp(data, "top") == 0)
+			placement = "top";
+		else if (g_ascii_strcasecmp(data, "bottom") == 0)
+			placement = "bottom";
+		else {
+			printformat(NULL, NULL, MSGLEVEL_CLIENTERROR,
+				    TXT_STATUSBAR_UNKNOWN_PLACEMENT, data);
+			error++;
+		}
+	}
 
-/* SYNTAX: STATUSBAR <name> TYPE window|root */
-static void cmd_statusbar_type(const char *data, void *server,
-			       void *item, CONFIG_NODE *node)
-{
-	if (g_ascii_strcasecmp(data, "window") == 0)
-		iconfig_node_set_str(node, "type", "window");
-        else if (g_ascii_strcasecmp(data, "root") == 0)
-		iconfig_node_set_str(node, "type", "root");
-	else {
-		printformat(NULL, NULL, MSGLEVEL_CLIENTERROR,
-			    TXT_STATUSBAR_UNKNOWN_TYPE, data);
+	visible = NULL;
+	data = g_hash_table_lookup(optlist, "visible");
+	if (data != NULL) {
+		if (g_ascii_strcasecmp(data, "always") == 0)
+			visible = "always";
+		else if (g_ascii_strcasecmp(data, "active") == 0)
+			visible = "active";
+		else if (g_ascii_strcasecmp(data, "inactive") == 0)
+			visible = "inactive";
+		else {
+			printformat(NULL, NULL, MSGLEVEL_CLIENTERROR,
+				    TXT_STATUSBAR_UNKNOWN_VISIBILITY, data);
+			error++;
+		}
 	}
-}
 
-/* SYNTAX: STATUSBAR <name> PLACEMENT top|bottom */
-static void cmd_statusbar_placement(const char *data, void *server,
-				    void *item, CONFIG_NODE *node)
-{
-	if (g_ascii_strcasecmp(data, "top") == 0)
-		iconfig_node_set_str(node, "placement", "top");
-        else if (g_ascii_strcasecmp(data, "bottom") == 0)
-		iconfig_node_set_str(node, "placement", "bottom");
-	else {
-		printformat(NULL, NULL, MSGLEVEL_CLIENTERROR,
-			    TXT_STATUSBAR_UNKNOWN_PLACEMENT, data);
+	if (error) {
+		cmd_params_free(free_arg);
+		return;
 	}
-}
 
-/* SYNTAX: STATUSBAR <name> POSITION <num> */
-static void cmd_statusbar_position(const char *data, void *server,
-                                   void *item, CONFIG_NODE *node)
-{
-	iconfig_node_set_int(node, "position", atoi(data));
+	node = sbar_node(name, TRUE);
```

Comment:
If i'm reading this right both add and modify call `sbar_node` with create=TRUE. Same thing with additem/modifyitem.

Reply:
oops